panic: interface conversion: interface {} is nil, not map[string]interface {}

Terraform Version
Terraform v1.5.1
on darwin_arm64
+ provider registry.terraform.io/datadog/datadog v3.8.1
--------------------------------
Terraform Configuration Files
...terraform config...
--------------------------------
Debug Output
The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).PlanResourceChange call. The plugin logs may contain more details.

goroutine 62 [running]:
github.com/terraform-providers/terraform-provider-datadog/datadog.deleteWidgetID(0x14000b4ef70, 0x1, 0x1)
github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_dashboard_json.go:87 +0x1fc
github.com/terraform-providers/terraform-provider-datadog/datadog.prepResource(0x1400029acf0, 0x2e4)
github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_dashboard_json.go:242 +0xf8
github.com/terraform-providers/terraform-provider-datadog/datadog.resourceDatadogDashboardJSON.func2(0x10111a700, 0x14000b4ee20, 0x9, 0x0)
github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_dashboard_json.go:56 +0xb8
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.schemaMap.diffString(0x140005686c0, 0x100e50242, 0x9, 0x140001f9680, 0x140002db3e8, 0x1013b1338, 0x140005ef680, 0x1002c0000, 0x0, 0x0)
github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.0/helper/schema/schema.go:1378 +0x510
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.schemaMap.diff(0x140005686c0, 0x100e50242, 0x9, 0x140001f9680, 0x140005ef600, 0x1013b1338, 0x140005ef680, 0x100323500, 0x0, 0x0)
github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.0/helper/schema/schema.go:960 +0x3c4
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.schemaMap.Diff(0x140005686c0, 0x1013acd78, 0x140006dd100, 0x140006cd930, 0x1400029aa50, 0x101383358, 0x1010b2600, 0x140005ecf00, 0x1400029aa00, 0x1400029aa50, ...)
github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.0/helper/schema/schema.go:525 +0x228
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).SimpleDiff(0x140001fa000, 0x1013acd78, 0x140006dd100, 0x140006cd930, 0x1400029aa50, 0x1010b2600, 0x140005ecf00, 0x0, 0x0, 0x0)
github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.0/helper/schema/resource.go:517 +0x68
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).PlanResourceChange(0x1400011a000, 0x1013acd78, 0x140006dd100, 0x14000b58cd0, 0x100e590d5, 0x12, 0x0)
github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.0/helper/schema/grpc_provider.go:703 +0x7d0
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).PlanResourceChange(0x14000596100, 0x1013ace20, 0x140006dd100, 0x140009d9110, 0x0, 0x0, 0x0)
github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:578 +0x338
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_PlanResourceChange_Handler(0x1012e5bc0, 0x14000596100, 0x1013ace20, 0x14000595e00, 0x140006b8c60, 0x0, 0x1013ace20, 0x14000595e00, 0x14000780700, 0x678)
github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:362 +0x1c8
google.golang.org/grpc.(*Server).processUnaryRPC(0x14000134700, 0x1013b51d8, 0x14000583500, 0x1400017a800, 0x140006163f0, 0x101c8cfc8, 0x0, 0x0, 0x0)
google.golang.org/grpc@v1.32.0/server.go:1194 +0x3e8
google.golang.org/grpc.(*Server).handleStream(0x14000134700, 0x1013b51d8, 0x14000583500, 0x1400017a800, 0x0)
google.golang.org/grpc@v1.32.0/server.go:1517 +0xa50
google.golang.org/grpc.(*Server).serveStreams.func1.2(0x14000034510, 0x14000134700, 0x1013b51d8, 0x14000583500, 0x1400017a800)
google.golang.org/grpc@v1.32.0/server.go:859 +0x94
created by google.golang.org/grpc.(*Server).serveStreams.func1
google.golang.org/grpc@v1.32.0/server.go:857 +0x1f8
--------------------------------
Expected Behavior
Plan should be successful
--------------------------------
Actual Behavior
Stack trace from the terraform-provider-datadog_v3.8.1 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 62 [running]:
github.com/terraform-providers/terraform-provider-datadog/datadog.deleteWidgetID(0x14000b4ef70, 0x1, 0x1)
github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_dashboard_json.go:87 +0x1fc
github.com/terraform-providers/terraform-provider-datadog/datadog.prepResource(0x1400029acf0, 0x2e4)
github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_dashboard_json.go:242 +0xf8
github.com/terraform-providers/terraform-provider-datadog/datadog.resourceDatadogDashboardJSON.func2(0x10111a700, 0x14000b4ee20, 0x9, 0x0)
github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_dashboard_json.go:56 +0xb8
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.schemaMap.diffString(0x140005686c0, 0x100e50242, 0x9, 0x140001f9680, 0x140002db3e8, 0x1013b1338, 0x140005ef680, 0x1002c0000, 0x0, 0x0)
github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.0/helper/schema/schema.go:1378 +0x510
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.schemaMap.diff(0x140005686c0, 0x100e50242, 0x9, 0x140001f9680, 0x140005ef600, 0x1013b1338, 0x140005ef680, 0x100323500, 0x0, 0x0)
github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.0/helper/schema/schema.go:960 +0x3c4
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.schemaMap.Diff(0x140005686c0, 0x1013acd78, 0x140006dd100, 0x140006cd930, 0x1400029aa50, 0x101383358, 0x1010b2600, 0x140005ecf00, 0x1400029aa00, 0x1400029aa50, ...)
github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.0/helper/schema/schema.go:525 +0x228
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).SimpleDiff(0x140001fa000, 0x1013acd78, 0x140006dd100, 0x140006cd930, 0x1400029aa50, 0x1010b2600, 0x140005ecf00, 0x0, 0x0, 0x0)
github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.0/helper/schema/resource.go:517 +0x68
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).PlanResourceChange(0x1400011a000, 0x1013acd78, 0x140006dd100, 0x14000b58cd0, 0x100e590d5, 0x12, 0x0)
github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.0/helper/schema/grpc_provider.go:703 +0x7d0
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).PlanResourceChange(0x14000596100, 0x1013ace20, 0x140006dd100, 0x140009d9110, 0x0, 0x0, 0x0)
github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:578 +0x338
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_PlanResourceChange_Handler(0x1012e5bc0, 0x14000596100, 0x1013ace20, 0x14000595e00, 0x140006b8c60, 0x0, 0x1013ace20, 0x14000595e00, 0x14000780700, 0x678)
github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:362 +0x1c8
google.golang.org/grpc.(*Server).processUnaryRPC(0x14000134700, 0x1013b51d8, 0x14000583500, 0x1400017a800, 0x140006163f0, 0x101c8cfc8, 0x0, 0x0, 0x0)
google.golang.org/grpc@v1.32.0/server.go:1194 +0x3e8
google.golang.org/grpc.(*Server).handleStream(0x14000134700, 0x1013b51d8, 0x14000583500, 0x1400017a800, 0x0)
google.golang.org/grpc@v1.32.0/server.go:1517 +0xa50
google.golang.org/grpc.(*Server).serveStreams.func1.2(0x14000034510, 0x14000134700, 0x1013b51d8, 0x14000583500, 0x1400017a800)
google.golang.org/grpc@v1.32.0/server.go:859 +0x94
created by google.golang.org/grpc.(*Server).serveStreams.func1
google.golang.org/grpc@v1.32.0/server.go:857 +0x1f8

Error: The terraform-provider-datadog_v3.8.1 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
--------------------------------
Steps to Reproduce
terraform init
terraform plan

--- a/datadog/resource_datadog_dashboard_json.go
+++ b/datadog/resource_datadog_dashboard_json.go
@@ -83,12 +83,14 @@ func resourceDatadogDashboardJSON() *schema.Resource {
 
 func deleteWidgetID(widgets []interface{}) {
 	for _, w := range widgets {
-		widget := w.(map[string]interface{})
-		def := widget["definition"].(map[string]interface{})
-		if def["type"] == "group" {
-			deleteWidgetID(def["widgets"].([]interface{}))
+		if widget, ok := w.(map[string]interface{}); ok {
+			if def, ok := widget["definition"].(map[string]interface{}); ok {
+				if def["type"] == "group" {
+					deleteWidgetID(def["widgets"].([]interface{}))
+				}
+				delete(widget, "id")
+			}
 		}
-		delete(widget, "id")
 	}
 }
 

--- a/datadog/resource_datadog_dashboard_json.go
+++ b/datadog/resource_datadog_dashboard_json.go
@@ -86,7 +86,9 @@ func deleteWidgetID(widgets []interface{}) {
 		if widget, ok := w.(map[string]interface{}); ok {
 			if def, ok := widget["definition"].(map[string]interface{}); ok {
 				if def["type"] == "group" {
-					deleteWidgetID(def["widgets"].([]interface{}))
+					if group, ok := def["widgets"].([]interface{}); ok {
+						deleteWidgetID(group)
+					}
 				}
 				delete(widget, "id")
 			}

